internal/buildscripts/packaging/tests/instrumentation/instrumentation_test.py

```diff
@@ -0,0 +1,252 @@
+    otelcol_bin_path = OTELCOL_BIN_DIR / otelcol_bin
+    assert os.path.isfile(otelcol_bin_path), f"{otelcol_bin_path} not found!"
+
+    pkg_path = get_package(distro, PKG_NAME, PKG_DIR, arch)
+    assert pkg_path, f"{PKG_NAME} package not found in {PKG_DIR}"
+    pkg_base = os.path.basename(pkg_path)
+
+    with run_distro_container(distro, dockerfile=get_dockerfile(distro), arch=arch) as container:
+        copy_file_into_container(container, COLLECTOR_CONFIG_PATH, "/test/config.yaml")
+        copy_file_into_container(container, pkg_path, f"/test/{pkg_base}")
+        copy_file_into_container(container, otelcol_bin_path, f"/test/{otelcol_bin}")
+        run_container_cmd(container, f"chmod a+x /test/{otelcol_bin}")
+
+        # install the instrumentation package
+        install_package(container, distro, f"/test/{pkg_base}")
+
+        # verify files were installed
+        run_container_cmd(container, f"test -f {JAVA_AGENT_PATH}")
+        run_container_cmd(container, f"test -f {DEFAULT_PROPERTIES_PATH}")
+        run_container_cmd(container, f"test -f {DEFAULT_CONF_PATH}")
+
+        verify_tomcat_instrumentation(container, distro, config, otelcol_path=f"/test/{otelcol_bin}")
+
+
+@pytest.mark.parametrize(
+    "distro",
+    [pytest.param(distro, marks=pytest.mark.deb) for distro in DEB_DISTROS]
+    + [pytest.param(distro, marks=pytest.mark.rpm) for distro in RPM_DISTROS],
+    )
+@pytest.mark.parametrize("arch", ["amd64", "arm64"])
+@pytest.mark.parametrize("config", ["env_vars", "properties_file"])
+def test_package_upgrade(distro, arch, config):
+    install_cmd = "sh /test/install.sh -- testing123 --realm test --without-fluentd " \
+                  "--collector-config /test/config.yaml --with-instrumentation --instrumentation-version 0.76.0"
+
+    pkg_path = get_package(distro, PKG_NAME, PKG_DIR, arch)
+    assert pkg_path, f"{PKG_NAME} package not found in {PKG_DIR}"
+    pkg_base = os.path.basename(pkg_path)
+
+    with run_distro_container(distro, dockerfile=get_dockerfile(distro), arch=arch) as container:
+        copy_file_into_container(container, COLLECTOR_CONFIG_PATH, "/test/config.yaml")
+        copy_file_into_container(container, INSTALLER_PATH, "/test/install.sh")
+        copy_file_into_container(container, pkg_path, f"/test/{pkg_base}")
+
+        # install the collector and an older version of the instrumentation package
+        run_container_cmd(container, install_cmd, env={"VERIFY_ACCESS_TOKEN": "false"}, timeout="10m")
+
+        # verify /etc/ld.so.preload and libsplunk.so were installed
+        run_container_cmd(container, "test -f /etc/ld.so.preload")
+        run_container_cmd(container, "test -f /usr/lib/splunk-instrumentation/libsplunk.so")
+
+        # upgrade the instrumentation package
+        install_package(container, distro, f"/test/{pkg_base}")
+
+        # verify /etc/ld.so.preload and libsplunk.so were deleted after upgrade
+        run_container_cmd(container, "test ! -f /etc/ld.so.preload")
```

Comment:
> Can we update the preload to contain another entry and ensure only it persists?

I'll update the test.

> I also wonder about the logic of deleting it outright even if we were the only content (unless it can be confirmed we created the file).

Currently, when the old `splunk-otel-auto-instrumentation` is uninstalled or replaced by this new package, `libsplunk.so` is first removed from `/etc/ld.so.preload`, and then the file is deleted only if it is empty (see [preuninstall.sh](https://github.com/signalfx/splunk-otel-collector/blob/main/instrumentation/packaging/fpm/preuninstall.sh)).